### PR TITLE
C2CX API integration tests

### LINKED ISSUE #81: C2CX API integration tests

### PR DESCRIPTION
**Pull request by Vavilen**
Fixes #81 
Completed test cases
- [x] get_orderbook
- [x] get_balance_summary
- [x] create_order
- [x] get_order_info
- [x] get_order_info_all
- [x] cancel_order
- [x] cancel_multiple
- [x] cancel_all
- [x] limit_buy
- [x] limit_sell
- [ ] market_buy
- [ ] market_sell
- [x] get_ticker
- [x] get_order_by_status
- [x] get_order_by_status_paged

### CODE REVIEW

**Comment by Vavilen** [comment]
@gz-c,  tests failed because there is no the config file. I will think out about ENV vars as the alternative for the config file.

**Comment by gz-c** [comment]
You can add a hidden file with test config such as ".test.cfg" and specify to use that in travis

File: exchange/c2cx/api.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (c *Client) GetOrderbook(symbol TradePair) (*Orderbook, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	//fmt.Println("data in GetOrderbook:\n" + string(data) + "\n===============\n\n")
```
Comment (gz-c) [inline exchange/c2cx/api.go]: delete

File: exchange/c2cx/api.go [hunk 2]
```diff
@@ -178,7 +178,7 @@ func (c *Client) GetBalanceSummary() (*BalanceSummary, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	//fmt.Println("data in GetBalanceSummary:\n" + string(data) + "\n===============\n\n")
```
Comment (gz-c) [inline exchange/c2cx/api.go]: delete

File: exchange/c2cx/c2cx.go [hunk 3]
```diff
@@ -428,8 +428,8 @@ func (br BalanceSummary) Spendable() Balances {
 
 // Orders includes []Order and page count if not empty
 type Orders struct {
-	Orders []Order `json:"orders"`
-	Page   int     `json:"page,omitempty"`
+	Orders    []Order `json:"orders"`
+	PageCount int     `json:"pageCount,omitempty"`
```
Comment (gz-c) [inline exchange/c2cx/c2cx.go]: page is different from pageCount, why is this changed?

File: exchange/order.go [hunk 4]
```diff
@@ -15,6 +15,12 @@ type MarketOrder struct {
 	Volume decimal.Decimal `json:"volume"`
 }
 
+// MarketOrder is a one order in stock
```
Comment (gz-c) [inline exchange/order.go]: what is this for?

**Review by gz-c — CHANGES_REQUESTED** [review]

File: exchange/order.go (hunk 4 above)
Comment (Vavilen) [reply exchange/order.go]: Moved it to the integration_test.go file. It is useful for unmarshaling orderbook response in a test.

File: exchange/c2cx/c2cx.go (hunk 3 above)
Comment (Vavilen) [reply exchange/c2cx/c2cx.go]: I thought that this param is useful to know it is necessary to know if there are more pages.
Added both params.

File: exchange/c2cx/cli/c2cx.go [hunk 5]
```diff
@@ -246,7 +246,7 @@ GetOrderByStatusPaged get all orders with given status for a given pagination pa
 					printErrorWithExit(err)
 				}
 				orders, pageCount, err := client.GetOrderByStatusPaged(c2cx.TradePair(tradePair), c2cx.OrderStatus(orderStatus), page)
-				handleResult(c2cx.Orders{Orders: orders, Page: pageCount}, err)
+				handleResult(c2cx.Orders{Orders: orders, PageCount: pageCount}, err)
```
Comment (gz-c) [inline exchange/c2cx/cli/c2cx.go]: Do you need to populate Page?

File: exchange/c2cx/integration_test.go [hunk 6]
```diff
@@ -1,18 +1,898 @@
...
+			if tc.errMessage != "" && err != nil {
+				require.Equal(t, tc.errMessage, string(output))
+				return
+			}
+			require.NoError(t, err)
+			var resp orderbookTest
+			err = resp.UnmarshalJSON(output)
+			require.NoError(t, err)
+			for _, bid := range resp.Bids {
+				require.True(t, bid.Volume.Cmp(decimal.New(0, 10)) > 0)
```
Comment (gz-c) [inline exchange/c2cx/integration_test.go]: could do `.GreaterThan(decimal.Zero)`

File: exchange/c2cx/integration_test.go [hunk 7]
```diff
@@ -1,18 +1,898 @@
...
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, tc.args...).CombinedOutput()
+			require.NoError(t, err)
+			fmt.Println(string(output))
+			var resp BalanceSummaryString
+			err = json.Unmarshal(output, &resp)
+			require.NoError(t, err)
+			require.True(t, len(resp.Balance) > 0)
+			require.True(t, len(resp.Frozen) > 0)
```
Comment (gz-c) [inline exchange/c2cx/integration_test.go:183]: can you check that the final calculated balance (which subtracts frozen) succeeds and have valid values (`>=0`)?

**Review by gz-c — CHANGES_REQUESTED** [review]

**Comment by gz-c** [comment]
@Vavilen some requested changes and the build fails

File: exchange/c2cx/.testConfig/.test.toml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[c2cx]
+key = "EBE87C06-D0A2-4942-B2D3-57AD6E7C242A"
+secret = "4D7683C9-BE09-44B8-AE32-08070A2F7540"
```
Comment (gz-c) [inline exchange/c2cx/.testConfig/.test.toml]: probably should not add this to the repo directly, we can add it to travis as a secret environment variable and add a travis step to write config file

also, the folder name should not be `.testConfig`, don't use mixed case for folder names

File: exchange/c2cx/api.go [hunk 9]
```diff
@@ -242,6 +240,7 @@ func (c *Client) CreateOrder(symbol TradePair, price, quantity decimal.Decimal,
 
 	var resp createOrderResponse
 	if err := json.Unmarshal(data, &resp); err != nil {
+		fmt.Println(string(data))
```
Comment (gz-c) [inline exchange/c2cx/api.go]: remove

File: exchange/c2cx/c2cx.go [hunk 10]
```diff
@@ -428,8 +428,9 @@ func (br BalanceSummary) Spendable() Balances {
 
 // Orders includes []Order and page count if not empty
 type Orders struct {
-	Orders []Order `json:"orders"`
-	Page   int     `json:"page,omitempty"`
+	Orders     []Order `json:"orders"`
+	Page       int     `json:"page,omitempty"`
+	PagesCount int     `json:"pagesCount,omitempty"`
```
Comment (gz-c) [inline exchange/c2cx/c2cx.go]: is it pageCount or pagesCount?

File: exchange/c2cx/cli/c2cx.go [hunk 11]
```diff
@@ -389,12 +389,18 @@ The public ticker API returns key pricing data for a give currency pair.
 }
 
 func init() {
-	user, err := user.Current()
-	if err != nil {
-		log.Panicf("failed to get the current user. err: %v", err)
+	var config string
+	var err error
+	if os.Getenv("TEST_EXCHANGE_CONFIG") == "" {
```
Comment (gz-c) [inline exchange/c2cx/cli/c2cx.go]: Do not have test specific code here.  Either name the envvar something like `EXCHANGE_API_CONFIG` or add a `-c <config>` CLI flag (https://github.com/skycoin/teller has an example of this)

**Review by gz-c — CHANGES_REQUESTED** [review]

File: exchange/c2cx/.testConfig/.test.toml (hunk 8 above)
Comment (Vavilen) [reply exchange/c2cx/.testConfig/.test.toml]: @gz-c , should I use travis functional (didn't find any) for saving data from a secret environment variable or run an external bash script from travis step, which:
1. reads data from a travis secret environment variable
2. saves data to the file with some path.
3. sets path to a config file to an environment variable EXVHANGE_API_CONFIG

File: exchange/c2cx/c2cx.go (hunk 10 above)
Comment (Vavilen) [reply exchange/c2cx/c2cx.go]: @gz-c , my bad. pageCount is correct.

File: exchange/c2cx/.testConfig/.test.toml (hunk 8 above)
Comment (gz-c) [reply exchange/c2cx/.testConfig/.test.toml]: I added these as envvars in the travis environment as C2CX_API_KEY and C2CX_API_SECRET, you can use them to write a config file during the travis setup

Comment (gz-c) [reply exchange/c2cx/.testConfig/.test.toml]: You should squash your commits into one to remove these keys from the commit history

**Review by gz-c — APPROVED** [review]

**Pull request merged**